src/client/CodeEditor/getOrCreateEditor.ts

```diff
@@ -195,6 +198,69 @@ export const getOrCreateEditor = ({
 
   extensions.push(rotationIndicator);
 
+/*
+  extensions.push(autocompletion({
+    override: [tsComplete]
+  }))*/
+
+  // displays autocompletes
+  async function tsComplete(ctx) {
+
+    const compilerOptions = {};
+    const fsMap = await tsvfs.createDefaultMapFromCDN(
```

Comment:
This approach creates one `fsMap` for each editor. We want to only ever create one single one of these.

I think the CodeMirror extension should just get the completions by passing messages into a Web Worker that contains a singleton `fsMap`.

There will be a whole part of this system that lives outside the CodeMirror extension, namely:

 * Something that sets up the Web Worker
 * Something that keeps the `fsMap` within the Web Worker in sync with our ShareDB-based file system